imap_processing/tests/glows/test_glows_l1a_data.py

```diff
@@ -361,3 +391,120 @@ def test_generate_status_data():
+            de.l0.ccsds_header.SRC_SEQ_CTR
+            == validation_data["seq_count_in_pkts_file"][index]
+        )
+        assert (
+            de.status_data.imap_sclk_last_pps
+            == validation_data["imap_sclk_last_pps"][index]
+        )
+        assert (
+            de.status_data.glows_sclk_last_pps
+            == validation_data["glows_sclk_last_pps"][index]
+        )
+        assert (
+            de.status_data.glows_ssclk_last_pps
+            == validation_data["glows_ssclk_last_pps"][index]
+        )
+        assert (
+            de.status_data.imap_sclk_next_pps
+            == validation_data["imap_sclk_next_pps"][index]
+        )
+        assert (
+            de.status_data.catbed_heater_active
+            == validation_data["catbed_heater_active"][index]
+        )
+        assert (
+            de.status_data.spin_period_valid
+            == validation_data["spin_period_valid"][index]
+        )
+        assert (
+            de.status_data.spin_phase_at_next_pps_valid
+            == validation_data["spin_phase_at_next_pps_valid"][index]
+        )
+        assert (
+            de.status_data.spin_period_source
+            == validation_data["spin_period_source"][index]
+        )
+        assert de.status_data.spin_period == validation_data["spin_period"][index]
+        assert (
+            de.status_data.spin_phase_at_next_pps
+            == validation_data["spin_phase_at_next_pps"][index]
+        )
+        assert (
+            de.status_data.number_of_completed_spins
+            == validation_data["number_of_completed_spins"][index]
+        )
+        assert (
+            de.status_data.filter_temperature
+            == validation_data["filter_temperature"][index]
+        )
+        assert de.status_data.hv_voltage == validation_data["hv_voltage"][index]
+        assert (
+            de.status_data.glows_time_on_pps_valid
+            == validation_data["glows_time_on_pps_valid"][index]
+        )
+        assert (
+            de.status_data.time_status_valid
+            == validation_data["time_status_valid"][index]
+        )
+        assert (
+            de.status_data.housekeeping_valid
+            == validation_data["housekeeping_valid"][index]
+        )
+        assert (
+            de.status_data.is_pps_autogenerated
+            == validation_data["is_pps_autogenerated"][index]
+        )
+        assert (
+            de.status_data.hv_test_in_progress
+            == validation_data["hv_test_in_progress"][index]
+        )
+        assert (
+            de.status_data.pulse_test_in_progress
+            == validation_data["pulse_test_in_progress"][index]
+        )
+        assert (
+            de.status_data.memory_error_detected
+            == validation_data["memory_error_detected"][index]
+        )
+
+        assert de.l0.LEN == validation_data["number_of_de_packets"][index]
+
+        validation_de = ast.literal_eval(validation_data["de_data"][index])
```

Comment:
Nice use of `literal_eval`!
Could you bring it up into the original read_csv as a converter? `read_csv(*, converters={"de_data": ast.literal_eval})`

Reply:
I did not realize this was something read_csv could do, nice!

Reply:
Also, a lazy use of literal_eval instead of adding new columns 😆